### PR TITLE
clientv3: disable client side grpc log

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/4811.

### CODE REVIEW

**Comment by gyuho** [comment]
I ran some simple code of `put` and `get` using `clientv3` and confirmed that this disables grpc logs when I killed one of the members.


Before 

```
2016/03/26 21:09:30 transport: http2Client.notifyError got notified that the client transport was broken EOF.
2016/03/26 21:09:31 grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:2379: getsockopt: connection refused"; Reconnecting to "localhost:2379"
2016/03/26 21:09:32 grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:2379: getsockopt: connection refused"; Reconnecting to "localhost:2379"
2016/03/26 21:09:32 grpc: Conn.transportMonitor exits due to: grpc: timed out trying to connect
2016/03/26 21:09:33 grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:2379: getsockopt: connection refused"; Reconnecting to "localhost:2379"
2016/03/26 21:09:35 grpc: Conn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 127.0.0.1:2379: getsockopt: connection refused"; Reconnecting to "localhost:2379"
```

After

```
[None]
```

File: clientv3/client.go [hunk 1]
```diff
@@ -26,8 +28,13 @@ import (
 	"golang.org/x/net/context"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/grpclog"
 )
 
+func init() {
```
Comment (xiang90) [inline clientv3/client.go]: user should be able to set logger themselves. (we by default disable gRPC logging)
probably add the logger interface into Config.

Comment (gyuho) [inline clientv3/client.go]: Agree. Will change. Thanks.

**Comment by gyuho** [comment]
@xiang90 Made the logger configurable. PTAL. Thanks.

File: clientv3/client.go [hunk 2]
```diff
@@ -66,6 +72,8 @@ type Config struct {
 
 	// TLS holds the client secure credentials, if any.
 	TLS *tls.Config
+
+	Logger grpclog.Logger
```
Comment (xiang90) [inline clientv3/client.go]: doc string?

File: clientv3/client.go [hunk 3]
```diff
@@ -180,6 +188,15 @@ func newClient(cfg *Config) (*Client, error) {
 	client.Watcher = NewWatcher(client)
 	client.Auth = NewAuth(client)
 	client.Maintenance = &maintenance{c: client}
+	if cfg.Logger == nil {
+		// same as default std logger
+		client.logger = log.New(os.Stderr, "", log.Ldate|log.Ltime)
```
Comment (xiang90) [inline clientv3/client.go]: i am not sure why we need to set client.logger in this case. if we want to set, it should be log.New(ioutil.Discard, "", 0).

Comment (gyuho) [inline clientv3/client.go]: Ok will just set it `log.New(ioutil.Discard, "", 0)`

**Comment by gyuho** [comment]
@xiang90 Fixed up and added godoc. PTAL. Thanks!

File: clientv3/client.go [hunk 4]
```diff
@@ -180,6 +189,15 @@ func newClient(cfg *Config) (*Client, error) {
 	client.Watcher = NewWatcher(client)
 	client.Auth = NewAuth(client)
 	client.Maintenance = &maintenance{c: client}
+	if cfg.Logger == nil {
+		// same as default std logger
+		client.logger = log.New(ioutil.Discard, "", 0)
```
Comment (xiang90) [inline clientv3/client.go]: the comment is not accurate. probably just remove the comment. we should always comment why not what. the code itself is the what part.

File: clientv3/client.go [hunk 5]
```diff
@@ -66,6 +71,10 @@ type Config struct {
 
 	// TLS holds the client secure credentials, if any.
 	TLS *tls.Config
+
+	// Logger defines client-sider log interface. If Logger is nil,
+	// logging will be ignored, including gRPC logs.
+	Logger grpclog.Logger
```
Comment (xiang90) [inline clientv3/client.go]: // Logger is the logger used by client library.

Comment (xiang90) [inline clientv3/client.go]: let's do a type define.

```go
type grpclog.Logger Logger
```

File: clientv3/client.go (hunk 4 above)
Comment (gyuho) [inline clientv3/client.go]: You are right. Just fixed. Thanks!

**Comment by xiang90** [comment]
lgtm

**Comment by gyuho** [comment]
thanks

**Pull request merged**